modules/decision/search_pattern.py

```diff
@@ -0,0 +1,98 @@
+"""
+Initiates/continues a search pattern when the drone cannot find a landing pad.
+"""
+
+from .. import decision_command
+from .. import odometry_and_time
+from math import tan, pi, cos, sin, ceil
+
+class SearchPattern:
+    """
+    Attributes:
+        camera_fov (float): 
+            Camera's field of view, measured in degrees. Use the smallest measurement available.
+            Is essentially assumed to be a circle
+        search_height (float): 
+            The altitude at which the search is conducted. This is used to calculate the pattern.
+            It does not make the drone go to this height
+        search_overlap (float): 
+            Overlap between passes, between 0 and 1. Recomended probably ~0.5, experiment with less.
+        current_position (OdometryAndTime):
+            The drone's current position.
+        acceptable_variance_squared (float):
+            The square of the acceptable distance from the target position.
+    """
+
+    def __init__(self,
+                 camera_fov: float,
+                 search_height: float,
+                 search_overlap: float,
+                 current_position: odometry_and_time.OdometryAndTime,
+                 acceptable_variance_squared: float):
+        
+        # Initialize the drone to the first position in the search pattern
+        self.current_ring = 0
+        self.current_pos_in_ring = 0
+        self.max_pos_in_ring = 0
+        self.search_radius = 0
+        self.acceptable_variance_squared = acceptable_variance_squared
+        
+        # Store the origin of the search and 
+        self.search_origin = current_position.odometry_data.position
+        self.target_posx = self.search_origin.north
+        self.target_posy = self.search_origin.east
+
+        # Calculate the gap between positions
+        self.search_width = 2 * search_height * tan((camera_fov * 180 / pi) / 2)
+        self.search_gap = self.search_width * (1 - search_overlap)
+    
+    def set_target_location(self):
+        """
+        Updates the target position to the next position in the search pattern.
+        """
+
+        # If it is done the current ring, move to the next ring. If not, next step in current ring
+        if self.current_pos_in_ring >= self.max_pos_in_ring:
+            self.current_ring += 1
+            self.current_pos_in_ring = 0
+            self.search_radius = self.search_gap * self.current_ring
+            self.max_pos_in_ring = (ceil(self.search_radius * 2 * pi / self.search_gap))
+            self.angle_between_positions = ((2 * pi) / (self.max_pos_in_ring + 1))
```

Comment:
Can you walk me through the math you did here? since the search pattern isn't actually a circle but an inscribed polygon, the circumference `self.search_radius * 2 * pi` would be bigger than the perimeter of the polygon right? This gives us too many search points - we're going in more than a circle.